### ISSUE TITLE
wrapping table cells/rows in custom components or html tags

### ISSUE THREAD

**Issue by glorp47** [AllenFang/react-bootstrap-table#1557]
So, I know we can use the dataFormatter to allow the text to show up as a link as well as using selectRow to use something like React Router to push a new URL. However, how can I make the entire cell or row (and not just the text contained within the cell) into a clickable link that can change depending on certain parameters?  Doesn't seem like it's currently possible and also seems like a relatively easy addition

**Comment by prajapati-parth** [comment]
Hello @glorp47, please use `onRowClick` option.

> how can I make the entire cell or row (and not just the text contained within the cell) into a clickable link that can change depending on certain parameters?

`onRowClick` has entire row object that you can use to decide where to redirect the link to.

Reference: page -> [this](https://allenfang.github.io/react-bootstrap-table/example.html#selection), section: Row Click

Thanks.

**Comment by glorp47** [comment]
Yeah, selectRow / onRowClick exists. But what about the ability to right click and open in a new window or tab? That functionality has to be hardcoded in and the user doesn't get to choose those options, they have to be set to execute via JavaScript.

**Comment by AllenFang** [comment]
@glorp47 you mean you want to right click event on the row? I think maybe I can support the `oncontextmenu` event on row for user. thanks

**Comment by prajapati-parth** [comment]
@glorp47 As another work around you can use anchor tags with `target='_blank'`. This will always open the link in new tab. Let me know if that solves your query.
Thanks.

**Comment by glorp47** [comment]
Well that just makes the text a link, and not the actual table cell / row itself (where you can right click and have the option of opening it regularly, or in a new tab or window).

I don't necessarily want to be locked into only having the onContextMenu event do just one specific thing, and I think making a custom right click menu might be overkill - however, if that's the only solution, that would be doable.

**Comment by AllenFang** [comment]
I guess `contextmenu` is only solution, because there's not way to detect the right click.
